graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+            """
+            # Hint: read the transformation comments inside-out, starting with split
+            vector_df = multi_string_df.select(
+                # transform outputs to an array so we convert to a DenseVector
+                array_to_vector(
+                    # After split, replace empty strings with 'NaN' and cast to Array<Float>
+                    F.transform(
+                        F.split(
+                            multi_string_df[self.multi_column], separator
+                        ),  # Split along the separator
+                        replace_empty_with_nan,
+                    ).cast(ArrayType(FloatType(), True))
+                ).alias(self.multi_column)
+            )
+            return vector_df
+
+        def vector_df_has_nan(vector_df: DataFrame, vector_col: str) -> bool:
+            """
+            Returns true if there exists at least one NaN value in `vector_df[vector_col]`
+            """
+            sum_vector_df = vector_df.select(Summarizer.mean(vector_df[vector_col]).alias("sum"))
+            sums_vector = sum_vector_df.take(1)[0]["sum"]
+            for val in sums_vector:
+                if np.isnan(val):
+                    return True
+            return False
+
+        # Convert the input column from either a delimited string or array to a Vector column
+        multi_col_type = input_df.schema.jsonValue()["fields"][0]["type"]
+        if multi_col_type == "string":
+            assert self.separator
+            vector_df = convert_multistring_to_vector_df(input_df, self.separator)
+        else:
+            vector_df = input_df.select(
+                array_to_vector(F.col(self.multi_column)).alias(self.multi_column)
+            )
+
+        if self.shared_imputation != "none":
+            # First we check if any NaN values exist in the input DF
+            # TODO: Replace column-level NaN search with row-level missing which should be faster.
+            input_has_nan = vector_df_has_nan(vector_df, self.multi_column)
+
+            if input_has_nan:
+                # If the input has NaN, check how values exist in the vectors
+                feat_row = input_df.take(1)[0]
+
+                len_array = len(feat_row[self.multi_column].split(self.original_separator))
+
+                # Doing the column splitting in memory can be expensive
+                if len_array > 50:
+                    logging.warning(
+                        "Attempting imputation on %d columns can lead to OOM errors", len_array
+                    )
+
+                # Splitting the vectors requires an array DF
+                if multi_col_type == "string":
+                    assert self.separator, "Separator must be provided for string-separated vectors"
+                    split_array_df = input_df.select(
+                        # After split, replace empty strings with 'NaN' and cast to Array<Float>
+                        F.transform(
```

Comment:
Can we call convert_multistring_to_vector_df here?

Reply:
Not quite, because that return a `DataFrame[DenseVector]` and in this case we want `DataFrame[ArrayType]`. I could modify that function though to support both return types.

Reply:
I think you can unify the implementation.